pset-iterators.js

```diff
@@ -184,6 +214,23 @@ console.log('Problem 7:')
 
 // Add your code below this line
 
+const previousPresidents = presidents.filter(president => {
+  if(president.left_office !== null){
+    return president
+  }
+})
+
+const shortTermPresidents = previousPresidents.filter(president => {
+  const tookOffice = parseInt(president.took_office.substring(0,4))
+  const leftOffice = parseInt(president.left_office.substring(0,4))
+
+  if(leftOffice - tookOffice < 4){
+    return president
+  }
+})
+
+console.log(shortTermPresidents)
+
```

Comment:
@rlou88 Good work here on this solution. I see where you were going with your solution and I have some suggestions.
- Currently, you are running a filter on `previousPresidents` but this will throw an error because `previousPresidents ` is undefined. You could replace `previousPresidents` with `presidents`.
-  Also since one of the presidents will return null we need to remove that president because if we run a substring on the value `null` this will throw an error. 

Take a look at this alternate solution [here](https://github.com/jsr20190826/hw-03-iterators-solution/blob/96bab20deba3ad116374a5827291a951faf5d555/pset-iterators.js#L201)